File: backend/app/models/payment.py
import enum
from sqlalchemy import Column, Float, DateTime, String
from app.db.base_class import Base
from app.utils.utils import tz_now


class PaymentStatus(str, enum.Enum):
    PENDING = "PENDING"
    COMPLETED = "COMPLETED"
    CANCELLED = "CANCELED"
    REVERSED = "REVERSED"


class PaymentMode(str, enum.Enum):
    ONLINE = "ONLINE"


class Payment(Base):
    amount = Column(Float(precision=2), nullable=False, default=0)
    created_at = Column(DateTime(), default=tz_now())
    status = Column(String(10), nullable=False)
    mode = Column(String(10), nullable=False)

File: market/lib/data/models/payment/payment.dart
// ignore_for_file: constant_identifier_names

import 'package:freezed_annotation/freezed_annotation.dart';

part 'payment.g.dart';
part 'payment.freezed.dart';

enum PaymentMode { ONLINE }

enum PaymentStatus { PENDING, COMPLETED, CANCELLED, REVERSED }

@freezed
class Payment with _$Payment {
  factory Payment({
    required int id,
    @JsonKey(name: "created_at") required DateTime createdAt,
    required PaymentMode mode,
    required PaymentStatus status,
    required double amount,
  }) = _Payment;
  factory Payment.fromJson(json) => _$$_PaymentFromJson(json);

}
